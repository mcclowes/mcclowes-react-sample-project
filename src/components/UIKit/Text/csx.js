import styled from "styled-components/macro";

export const Header = styled.h1`
  color: ${(props) => props.theme.colors.black};
  font-size: 1.2em;
  font-family: "Cantarell", sans-serif;
`;

Header.defaultProps = {
  theme: {
    colors: {
      black: "#333333",
    },
  },
};

export const Paragraph = styled.p`
  color: ${(props) => props.theme.colors.black};
  font-family: "Cantarell", sans-serif;
`;

Paragraph.defaultProps = {
  theme: {
    colors: {
      black: "#333333",
    },
  },
};

export const Meta = styled.p`
  margin: 0;
  font-size: 0.8em;
  color: ${(props) => props.theme.colors.gray};
  font-family: "Cantarell", sans-serif;
`;

Meta.defaultProps = {
  theme: {
    colors: {
      gray: "#aaaaaa",
    },
  },
};
