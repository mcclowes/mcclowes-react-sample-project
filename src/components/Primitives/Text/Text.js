import React from "react";
import { Header, Paragraph, Meta } from "./csx";
import PropTypes from "prop-types";

const Text = (props) => {
  const { children } = props;

  return <Paragraph>{children}</Paragraph>;
};

Text.propTypes = {
  children: PropTypes.any,
};

Text.Header = Header;
Text.Paragraph = Paragraph;
Text.Meta = Meta;

export default Text;
