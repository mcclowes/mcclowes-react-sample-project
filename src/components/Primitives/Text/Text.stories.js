import React from "react";
import Text from "./Text";

export default { title: "Text" };

export const text = () => {
  return <Text>Paragraph</Text>;
};

export const header = () => {
  return <Text.Header>Header</Text.Header>;
};

export const paragraph = () => {
  return <Text.Paragraph>Paragraph</Text.Paragraph>;
};

export const meta = () => {
  return <Text.Meta>Meta</Text.Meta>;
};
