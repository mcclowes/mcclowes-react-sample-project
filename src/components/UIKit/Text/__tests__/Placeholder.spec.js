import Text from "../Text";
import React from "react";
import { shallow } from "enzyme";

const defaultProps = {};

const renderComponent = (newProps, textType = "Paragraph") => {
  const props = {
    ...defaultProps,
    ...newProps,
  };

  const Component = Text[textType];

  return shallow(<Component {...props}>Some text goes here</Component>);
};

let wrapper;

describe("Text", () => {
  it("renders default component", () => {
    wrapper = renderComponent();

    expect(wrapper).toMatchSnapshot();
  });

  describe("Header", () => {
    it("renders default component", () => {
      wrapper = renderComponent();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Text", () => {
    it("renders default component", () => {
      wrapper = renderComponent();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Text", () => {
    it("renders default component", () => {
      wrapper = renderComponent();

      expect(wrapper).toMatchSnapshot();
    });
  });
});
