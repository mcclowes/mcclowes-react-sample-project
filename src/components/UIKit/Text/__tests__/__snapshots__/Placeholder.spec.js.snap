// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Header renders default component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {
        "theme": Object {
          "colors": Object {
            "black": "#333333",
          },
        },
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "csx__Paragraph-sc-5c5e0o-1",
        "isStatic": false,
        "rules": Array [
          "color:",
          [Function],
          ";font-family:\\"Cantarell\\",sans-serif;",
        ],
      },
      "displayName": "csx__Paragraph",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "csx__Paragraph-sc-5c5e0o-1",
      "target": "p",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "colors": Object {
        "black": "#333333",
      },
    }
  }
>
  Some text goes here
</StyledComponent>
`;

exports[`Text Text renders default component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {
        "theme": Object {
          "colors": Object {
            "black": "#333333",
          },
        },
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "csx__Paragraph-sc-5c5e0o-1",
        "isStatic": false,
        "rules": Array [
          "color:",
          [Function],
          ";font-family:\\"Cantarell\\",sans-serif;",
        ],
      },
      "displayName": "csx__Paragraph",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "csx__Paragraph-sc-5c5e0o-1",
      "target": "p",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "colors": Object {
        "black": "#333333",
      },
    }
  }
>
  Some text goes here
</StyledComponent>
`;

exports[`Text Text renders default component 2`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {
        "theme": Object {
          "colors": Object {
            "black": "#333333",
          },
        },
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "csx__Paragraph-sc-5c5e0o-1",
        "isStatic": false,
        "rules": Array [
          "color:",
          [Function],
          ";font-family:\\"Cantarell\\",sans-serif;",
        ],
      },
      "displayName": "csx__Paragraph",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "csx__Paragraph-sc-5c5e0o-1",
      "target": "p",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "colors": Object {
        "black": "#333333",
      },
    }
  }
>
  Some text goes here
</StyledComponent>
`;

exports[`Text renders default component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {
        "theme": Object {
          "colors": Object {
            "black": "#333333",
          },
        },
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "csx__Paragraph-sc-5c5e0o-1",
        "isStatic": false,
        "rules": Array [
          "color:",
          [Function],
          ";font-family:\\"Cantarell\\",sans-serif;",
        ],
      },
      "displayName": "csx__Paragraph",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "csx__Paragraph-sc-5c5e0o-1",
      "target": "p",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "colors": Object {
        "black": "#333333",
      },
    }
  }
>
  Some text goes here
</StyledComponent>
`;
